File: 0007-reverse-integer/0007-reverse-integer.java
class Solution {
    public int reverse(int x) {
        boolean isNegative = false;
        if(x<0){
            isNegative = true;
        }
        x = Math.abs(x);
        
        // get the reverse
        int reverse = 0;
        while(x!=0){
            int digit = x % 10;
            
            // check if outofbounds
            if (reverse < Integer.MIN_VALUE/10 || reverse > Integer.MAX_VALUE/10){return 0;}

            reverse = reverse*10 + digit;
            x /= 10;
        }
        
        // check if negative
        if(isNegative){
            reverse = reverse * (-1);
        }

        
        return reverse;
    }
}

File: 0007-reverse-integer/0007-reverse-integer.py
class Solution:
    def reverse(self, x: int) -> int:
        INT_MIN, INT_MAX = -2**31, 2**31-1
        
        
        str_x = str(x)
        negative = False
        
        if str_x[0] == '-':
            negative = True
            str_x = str_x[1:]
        
        reversed_str = str_x[::-1]
        reversed_int = int(reversed_str)
        
        if negative:
            reversed_int = -reversed_int
        
        if reversed_int < INT_MIN or reversed_int > INT_MAX:
            return 0
        
        return reversed_int